packages/api/src/command/medical/patient/settings/create-patient-settings.ts

```diff
@@ -104,21 +138,51 @@ export async function upsertPatientSettingsForCx({
 }: PatientSettingsUpsertForCxProps): Promise<PatientSettingsUpsertResults> {
   const { log } = out(`upsertPatientSettingsForCx - cx ${cxId}`);
 
-  const patientIds = await getPatientIds({ cxId, facilityId });
-
-  if (!patientIds.length) {
+  const totalPatients = await getPatientsCount({ cxId, facilityId });
+  if (totalPatients === 0) {
     log(`No patients found for cx ${cxId}`);
     return { patientsFoundAndUpdated: 0 };
   }
 
-  const patientsFoundAndUpdated = await upsertPatientSettings({
-    patientIds,
-    cxId,
-    settings,
-  });
+  let processedTotal = 0;
+  let failedTotal = 0;
+  let fromItem: string | undefined;
+
+  do {
+    const { currPageItems, nextPageItemId } = await getPaginationItems(
+      { count: BATCH_SIZE, ...(fromItem ? { fromItem } : {}) },
+      pagination => getPatients({ cxId, facilityId, pagination }),
```

Comment:
Before we were only loading the patient IDs, now we're loading the whole patients.

Couldn't we just keep the previous version that gets all the patient IDs and then batches only the upserts?

IMO the issue was not with loading a lot of IDs from the DB, but with the upsert of those into the DB.

For reference, I populated an array w/ 1MM UUIDs in 5 seconds on local.